factfinder/main.py

```diff
@@ -120,74 +224,107 @@ def create_variable(self, pff_variable) -> Variable:
-            client = self.c.acs5dp
-        elif v.source == "subject":
-            client = self.c.acs5st
-        elif v.source == "decennial":
-            client = self.c.sf1
-        else:
-            client = self.c.acs5
+        client = self.client_options.get(v.source, self.c.acs5)
 
         # 3. pulling data from census site
-        if geotype == "NTA":
-            df = self.aggregate_horizontal(client, v, "tract")
-            df = self.aggregate_vertical(df, from_geotype="tract", to_geotype="NTA")
-        elif geotype == "cd":
-            df = self.aggregate_horizontal(client, v, "tract")
-            df = self.aggregate_vertical(df, from_geotype="tract", to_geotype="cd")
-        elif geotype == "cd_fp_500":
-            if v.source == "decennial":
-                df = self.aggregate_horizontal(client, v, "block")
-                df = self.aggregate_vertical(
-                    df, from_geotype="block", to_geotype="cd_fp_500"
-                )
-            else:
-                df = self.aggregate_horizontal(client, v, "block group")
-                df = self.aggregate_vertical(
-                    df, from_geotype="block group", to_geotype="cd_fp_500"
-                )
-        elif geotype == "cd_fp_100":
-            if v.source == "decennial":
-                df = self.aggregate_horizontal(client, v, "block")
-                df = self.aggregate_vertical(
-                    df, from_geotype="block", to_geotype="cd_fp_100"
-                )
-            else:
-                df = self.aggregate_horizontal(client, v, "block group")
-                df = self.aggregate_vertical(
-                    df, from_geotype="block group", to_geotype="cd_fp_100"
+        from_geotype, aggregate_vertical = self.get_aggregate_vertical(
+            v.source, geotype
+        )
+        df = self.aggregate_horizontal(client, v, from_geotype)
+        df = aggregate_vertical(df)
+        return df
+
+    def calculate_c_e_m_p_z(self, v:Variable, geotype:str) -> pd.DataFrame:
+        """
+        this function will calculate e, m first, then based on if the 
+        variable is a median variable or base variable, it would calculate
+        p and z accordingly. Note that c calculation is the same for all variables
+        """
+        output_fields = [
+            "census_geoid",
+            "pff_variable",
+            "geotype",
+            "c",
+            "e",
+            "m",
+            "p",
+            "z",
+        ]
+        # If pff_variable is a median variable, then we would need
+        # to calculate using calculate_median_e_m for aggregated geography
+        # there's no need to calculate p, z for median variables
+        if v.pff_variable in self.median_variables:
+            df = (
+                self.calculate_median_e_m(v.pff_variable, geotype)
+                if geotype in self.aggregated_geography
+                else self.calculate_e_m(v.pff_variable, geotype)
```

Comment:
If `pff_variable` is a median and `geotype` is not in one of the aggregate geographies then we use the non-median function. Once this gets called, the the list of census variables associated with `pff_variable` in metadata automatically get summed. Do we need to build in a check that the list of variables in metadata for a median variable only has one thing to make sure the horizontal aggregation just passes through the downloaded value?

Reply:
Unless we want to keep it as is, assume our metadata is accurate, and leave options open for summing medians if a future use-case comes up.

Reply:
I'm not sure, the current set up does assume a lot of things about the metadata